### PR TITLE
[TT-2216] - Add support for custom signature in query parameters

### PR DESCRIPTION
**Pull request by gmaiztegi**
## Description
This PR adds `use_param` and `param` to the signature configuration, in order to be able to pass the signature as a query parameter (I would appreciate some help to know how to correctly document these new fields). These parameters behave very similar to those regarding the token.

## Related Issue

https://github.com/TykTechnologies/tyk/issues/3511

## Motivation and Context

https://github.com/TykTechnologies/tyk/commit/bde50bceb33dc7c98ff47cdf0d1f6f33dd240720 added support for a Mashery-style signature validation, but it only supports getting the signature from a header. Mashery supports passing the signature in a query parameter, so it would be great to support this feature and mimic this behaviour.

## How This Has Been Tested

I have to say I'm super new to the Golang community so I tested this only with the test added to this PR, so I would appreciate some feedback while I build the Docker image and test it inside Kubernetes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [X] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [X] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [X] My change requires a change to the documentation.
  - [X] If you've changed APIs, describe what needs to be updated in the documentation.
  - [ ] If new config option added, ensure that it can be set via ENV variable
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [X] Check your code additions will not fail linting checks:
  - [X] `go fmt -s`
  - [X] `go vet`

### CODE REVIEW

**Comment by christtyk** [comment]
@buger can you review pls :pray:

File: apidef/api_definitions.go [hunk 1]
```diff
@@ -477,6 +477,8 @@ type AuthConfig struct {
 type SignatureConfig struct {
 	Algorithm        string `mapstructure:"algorithm" bson:"algorithm" json:"algorithm"`
 	Header           string `mapstructure:"header" bson:"header" json:"header"`
+	UseParam         bool   `mapstructure:"use_param" bson:"use_param" json:"use_param"`
+	Param            string `mapstructure:"param" bson:"param" json:"param"`
```
Comment (buger) [inline apidef/api_definitions.go:481]: Lets make it consistent with another config options we have https://github.com/TykTechnologies/tyk/blob/76103a37c1a082c167d085bfee5d8fdde378d0ab/apidef/api_definitions.go#L468

Just name it ParamName  `param_name`.

**Review by buger — CHANGES_REQUESTED** [review]

**Comment by buger** [comment]
Added a small comment, otherwise looks solid! 
@chris-fewtrell pls help create Jira ticket and assign to next release.

File: apidef/api_definitions.go (hunk 1 above)
Comment (gmaiztegi) [reply apidef/api_definitions.go:481]: As the header parameter is just `header` and not `header_name`, I was following that naming. But you are right to make it equal to the `param_name`.

**Review by furkansenharputlu — APPROVED** [review]

**Comment by christtyk** [comment]
hey @gmaiztegi - this will probably get pushed out in 3.3 - will let you know. Want to drop me your address, we'd like to ping you some swag for the contribution! chris@tyk.io

**Comment by buger** [comment]
Merged via https://github.com/TykTechnologies/tyk/pull/3662 (essentially it is your change, with fixed conflicts).

**Pull request closed without merging**